### PR TITLE
Support v2 Linter API

### LINKED ISSUE #17: Support v2 Linter API
In the [v2.3.0 release](https://github.com/steelbrain/linter/releases/tag/v2.3.0) of `linter` support for the v1 API was removed, breaking semver. This means that this provider is no longer supported by the current version of `linter`.

To get this provider working again it will need to be upgrade to the v2 API, a guide for accomplishing that can be found here: [Upgrading to Standard Linter v2](https://github.com/steelbrain/linter/blob/v2.3.0/docs/guides/upgrading-to-standard-linter-v2.md).

Until this is fixed in order to use this provider you will need to downgrade to the last version of `linter` that supports the v1 API, you can do this with `apm install linter@2.2.0`.

### PR DESCRIPTION
**Pull request by vzamanillo**
Fixes #17 

plus configSchema in package.json and ensure severity

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,75 @@
...
-    "atom": ">0.50.0"
-  },
-  "providedServices": {
-    "linter": {
-      "versions": {
-        "1.0.0": "provideLinter"
-      }
-    }
-  }
+	"name": "linter-swiftc",
```
Comment (Arcanemagus) [inline package.json]: 2 spaces for indentation instead of tabs please 😉.

File: package.json [hunk 2]
```diff
@@ -1,19 +1,75 @@
...
+	"activationCommands": [],
```
Comment (Arcanemagus) [inline package.json]: 🔥 This line isn't doing anything, no reason to have it in here.

File: package.json [hunk 3]
```diff
@@ -1,19 +1,75 @@
...
+	"main": "./lib/linter-swiftc",
+	"version": "2.0.0",
+	"description": "Lint Swift using swiftc",
+	"repository": "https://github.com/AtomLinter/linter-swiftc",
+	"license": "MIT",
+	"engines": {
+		"atom": ">0.50.0"
```
Comment (Arcanemagus) [inline package.json]: Add `<2.0.0` on the end, just to be safe!

File: package.json [hunk 4]
```diff
@@ -1,19 +1,75 @@
...
+	},
+	"configSchema": {
+		"compilerExecPath": {
+			"default": "swiftc",
+			"description": "The path of the compiler executable, with binary name. By default will use `swiftc` from your path.",
+			"type": "string"
+		}
+	},
+	"dependencies": {
+		"fs-plus": "^3.0.0",
```
Comment (Arcanemagus) [inline package.json]: 🔥 I don't see this being used in the code anywhere.

File: package.json [hunk 5]
```diff
@@ -1,19 +1,75 @@
...
+		"atom-linter": "^10.0.0",
+		"atom-package-deps": "^5.0.0"
```
Comment (Arcanemagus) [inline package.json]: 🔥 I don't see this being used in the code anywhere.

File: package.json (hunk 1 above)
Comment (vzamanillo) [reply package.json]: omg!

**Pull request merged**